pkg/controller/controller_broker.go

```diff
@@ -544,7 +544,9 @@ func (c *controller) reconcileServiceClassFromServiceBrokerCatalog(broker *v1alp
 
 // reconcileServicePlan reconciles a ServicePlan after the
 // ServiceClass's catalog has been re-listed.
-func (c *controller) reconcileServicePlan(servicePlan *v1alpha1.ServicePlan) error {
+func (c *controller) reconcileServicePlan(broker *v1alpha1.ServiceBroker, servicePlan *v1alpha1.ServicePlan) error {
+	servicePlan.Spec.ServiceBrokerName = broker.Name
```

Comment:
do we want to update the strategy to not allow the user to set this field? Akin to how we do not allow the user to set the refs

Reply:
I don't think we should do that because you have to be able to set it on create.

We could make it so that you cannot update this field - but I'd prefer to do that in a follow-up

Reply:
Can you create an issue to track that followup work?

Reply:
> I don't think we should do that because you have to be able to set it on create.

@pmorie have you? can't you just set it on the server side by walking through plan->serviceClass->broker? Allowing user to set it seems error prone, and can be different from the actual broker, right?